src/java.base/share/classes/com/sun/crypto/provider/AES_Crypt.java

```diff
@@ -0,0 +1,1409 @@
+            a0 = T0[ti0 >>> 24] ^ T1[(ti1 >> 16) & 0xFF]
+                    ^ T2[(ti2 >> 8) & 0xFF] ^ T3[ti3 & 0xFF] ^ K[w];
+            a1 = T0[ti1 >>> 24] ^ T1[(ti2 >> 16) & 0xFF]
+                    ^ T2[(ti3 >> 8) & 0xFF] ^ T3[ti0 & 0xFF] ^ K[w + 1];
+            a2 = T0[ti2 >>> 24] ^ T1[(ti3 >> 16) & 0xFF]
+                    ^ T2[(ti0 >> 8) & 0xFF] ^ T3[ti1 & 0xFF] ^ K[w + 2];
+            a3 = T0[ti3 >>> 24] ^ T1[(ti0 >> 16) & 0xFF]
+                    ^ T2[(ti1 >> 8) & 0xFF] ^ T3[ti2 & 0xFF] ^ K[w + 3];
+
+            ti0 = T0[a0 >>> 24] ^ T1[(a1 >> 16) & 0xFF]
+                    ^ T2[(a2 >> 8) & 0xFF] ^ T3[a3 & 0xFF] ^ K[w + 4];
+            ti1 = T0[a1 >>> 24] ^ T1[(a2 >> 16) & 0xFF]
+                    ^ T2[(a3 >> 8) & 0xFF] ^ T3[a0 & 0xFF] ^ K[w + 5];
+            ti2 = T0[a2 >>> 24] ^ T1[(a3 >> 16) & 0xFF]
+                    ^ T2[(a0 >> 8) & 0xFF] ^ T3[a1 & 0xFF] ^ K[w + 6];
+            ti3 = T0[a3 >>> 24] ^ T1[(a0 >> 16) & 0xFF]
+                    ^ T2[(a1 >> 8) & 0xFF] ^ T3[a2 & 0xFF] ^ K[w + 7];
+            w += 8;
+        }
+        if (rounds > AES_192_ROUNDS) {
+            a0 = T0[ti0 >>> 24] ^ T1[(ti1 >> 16) & 0xFF]
+                    ^ T2[(ti2 >> 8) & 0xFF] ^ T3[ti3 & 0xFF] ^ K[w];
+            a1 = T0[ti1 >>> 24] ^ T1[(ti2 >> 16) & 0xFF]
+                    ^ T2[(ti3 >> 8) & 0xFF] ^ T3[ti0 & 0xFF] ^ K[w + 1];
+            a2 = T0[ti2 >>> 24] ^ T1[(ti3 >> 16) & 0xFF]
+                    ^ T2[(ti0 >> 8) & 0xFF] ^ T3[ti1 & 0xFF] ^ K[w + 2];
+            a3 = T0[ti3 >>> 24] ^ T1[(ti0 >> 16) & 0xFF]
+                    ^ T2[(ti1 >> 8) & 0xFF] ^ T3[ti2 & 0xFF] ^ K[w + 3];
+
+            ti0 = T0[a0 >>> 24] ^ T1[(a1 >> 16) & 0xFF]
+                    ^ T2[(a2 >> 8) & 0xFF] ^ T3[a3 & 0xFF] ^ K[w + 4];
+            ti1 = T0[a1 >>> 24] ^ T1[(a2 >> 16) & 0xFF]
+                    ^ T2[(a3 >> 8) & 0xFF] ^ T3[a0 & 0xFF] ^ K[w + 5];
+            ti2 = T0[a2 >>> 24] ^ T1[(a3 >> 16) & 0xFF]
+                    ^ T2[(a0 >> 8) & 0xFF] ^ T3[a1 & 0xFF] ^ K[w + 6];
+            ti3 = T0[a3 >>> 24] ^ T1[(a0 >> 16) & 0xFF]
+                    ^ T2[(a1 >> 8) & 0xFF] ^ T3[a2 & 0xFF] ^ K[w + 7];
+            w += 8;
+        }
+
+        a0 = T2[ti0 >>> 24] & 0xFF000000
+                ^ T3[(ti1 >> 16) & 0xFF] & 0xFF0000
+                ^ T0[(ti2 >> 8) & 0xFF] & 0xFF00
+                ^ T1[ti3 & 0xFF] & 0xFF ^ K[w];
+        a1 = T2[ti1 >>> 24] & 0xFF000000
+                ^ T3[(ti2 >> 16) & 0xFF] & 0xFF0000
+                ^ T0[(ti3 >> 8) & 0xFF] & 0xFF00
+                ^ T1[ti0 & 0xFF] & 0xFF ^ K[w+1];
+        a2 = T2[ti2 >>> 24] & 0xFF000000
+                ^ T3[(ti3 >> 16) & 0xFF] & 0xFF0000
+                ^ T0[(ti0 >> 8) & 0xFF] & 0xFF00
+                ^ T1[ti1 & 0xFF] & 0xFF ^ K[w+2];
+        a3 = T2[ti3 >>> 24] & 0xFF000000
+                ^ T3[(ti0 >> 16) & 0xFF] & 0xFF0000
+                ^ T0[(ti1 >> 8) & 0xFF] & 0xFF00
+                ^ T1[ti2 & 0xFF] & 0xFF ^ K[w+3];
```

Comment:
Here you always use the last 4 elements of `K`, so you can just use `w = K.length - 4` and no need to keep tracking it in the earlier 2 blocks.